GitUI/ColorSchemeController.cs

```diff
@@ -0,0 +1,247 @@
+                .GetProperty("ThreadData", BindingFlags.Static | BindingFlags.NonPublic);
+
+            SystemBrushesKey = typeof(SystemBrushes)
+                .GetField("SystemBrushesKey", BindingFlags.Static | BindingFlags.NonPublic)
+                .GetValue(null);
+
+            SystemPensKey = typeof(SystemPens)
+                .GetField("SystemPensKey", BindingFlags.Static | BindingFlags.NonPublic)
+                .GetValue(null);
+        }
+
+        public static event Action SystemColorsChanging;
+
+        public static event Action SystemColorsChanged;
+
+        private IDictionary ThreadData =>
+            (IDictionary)_threadDataProperty.GetValue(null, null);
+
+        private object SystemBrushesKey { get; }
+
+        private object SystemPensKey { get; }
+
+        private HashSet<KnownColor> KnownColors { get; } = new HashSet<KnownColor>(
+            new[]
+            {
+                SystemColors.Control,
+                SystemColors.ControlDark,
+                SystemColors.ControlDarkDark,
+
+                SystemColors.ControlLight,
+                SystemColors.ControlLightLight,
+
+                SystemColors.Menu,
+                SystemColors.MenuText,
+
+                SystemColors.ControlText,
+
+                SystemColors.ButtonFace, // menu gradient
+                SystemColors.ButtonShadow, // menu border
+                SystemColors.ButtonHighlight,
+
+                SystemColors.Window,
+                SystemColors.WindowText,
+                SystemColors.GrayText,
+
+                SystemColors.HotTrack,
+                SystemColors.Highlight,
+                SystemColors.HighlightText,
+
+                SystemColors.ActiveCaption,
+                SystemColors.GradientActiveCaption,
+
+                SystemColors.InactiveCaption,
+                SystemColors.GradientInactiveCaption,
+
+                SystemColors.ActiveBorder,
+                SystemColors.Info,
+                SystemColors.InfoText,
+            }.Select(_ => _.ToKnownColor()));
+
+        private int[] OriginalColors { get; }
+
+        private IReadOnlyDictionary<int, int> KnownOriginalColors { get; }
+
+        private static void FireColorsChangedEvents()
+        {
+            SystemColorsChanging?.Invoke();
+            SystemColorsChanged?.Invoke();
+        }
+
+        public bool LoadDarkColorScheme() =>
+            Load(Properties.Resources.dark_colors);
+
+        private bool Load(string serialized, bool quiet = false)
+        {
+            if (TryDeserialize(serialized, out var deserialized))
+            {
+                Load(deserialized);
+                return true;
+            }
+
+            return false;
+
+            bool TryDeserialize(string input, out IReadOnlyDictionary<int, int> result)
+            {
+                result = null;
+                var lines = input.Split(
+                    new[] { Environment.NewLine },
```

Comment:
Suggest to split on "\n" too (required for me).

Reply:
ok